lib/chat_message_move_service.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+##
+# Used to move chat messages from a chat channel to some other
+# location.
+#
+# Channel -> Channel:
+# -------------------
+#
+# Messages are sometimes misplaced and must be moved to another channel. For
+# now we only support moving messages between public channels, handling the
+# permissions and membership around moving things in and out of DMs is a little
+# much for V1.
+#
+# The original messages will be deleted, and then similar to PostMover in core,
+# all of the references associated to a chat message (e.g. reactions, bookmarks,
+# notifications, revisions, mentions, uploads) will be updated to the new
+# message IDs via a moved_chat_messages temporary table.
+class ChatMessageMoveService
+  class NoMessagesFound < StandardError; end
+  class InvalidChannel < StandardError; end
+
+  def initialize(acting_user:, source_channel:, message_ids:)
+    @source_channel = source_channel
+    @acting_user = acting_user
+    @source_message_ids = message_ids
+    @source_messages = find_messages(@source_message_ids, source_channel)
+    @ordered_source_message_ids = @source_messages.map(&:id)
+  end
+
+  def move_to_channel(destination_channel)
```

Comment:
I wanted it this way because I figured that this class could eventually also be used for `move_to_topic` when we introduce that again -- one class many targets (even if eventually we refactor to have this parent class that calls out to a `MoveMessagesToChannel` and `MoveMessagesToTopic` classes